### ISSUE TITLE
Scrape price history

### ISSUE THREAD

**Issue by glenviewjeff** [hahnicity/scrapezillow#2]
My main motivation for this tool is to get at data that isn't available via the official API.  Particularly, the price history.  I plan to try to use this data to see if I can predict a length of time needed to sell using Zestimate and asking price as inputs.  If this is too much work, I can try to modify the script; my main problem is that I don't know how to run directly from sources.  In particular, the entire history in the table for the "date," "event," and "price" fields are all I'm looking for.

Thanks for all the help!

**Comment by glenviewjeff** [comment]
Just an update: This looks like it will be a huge pain in the ass.  The price/tax history isn't embedded within the source; it makes an Ajax call to indirectly pull in source.

**Comment by glenviewjeff** [comment]
Adding some detail (for myself or anyone else who wants to pick this up later) The Ajax call that pulls in the price history data is in a Javascript block.  The function call has a label that says "jsModule:"z-hdp-price-history", which could be used as part of a regex.  For [this page](http://www.zillow.com/homedetails/336-Greenleaf-Ave-Wilmette-IL-60091/3376784_zpid/) the full call in the script looks like this:

i.load({ajaxURL:"/AjaxRender.htm?encparams=1~2052912012659845440~NXf8Vr-y_xHr7zd_sj-VTDuJFJ2QMWKzR9I1kBqk91GuPhsvIpx08Yyxvmm0gaoLXs9K5ukPvlrfhWLOHMKmiypTyfyc9lZNLxgUq-Z2ltSYkDfanqHVXHFlTHRKY1UFBRX_CaHfzVw_bjePOKwKjk9S9Mt4teFQVjfG_yskNd1DqLkohYx1G78ckdtpI4MSbaTjuVKK2tPIiVDFjrJ7iV-PcPdVfeLv5qqiUw0Eb4voie71y2YuZ6MGQXPg-IqpWIfZYstBYEQ0GkK9__RLPBYnHF5hfXYlBkXEes6q-bhhFwKgHl2f-r0bnzUpfxPF1ltFc_SAtwsmHU9gV9zLnQ==&rwebid=7113216&rhost=1",jsModule:"z-hdp-price-history",phaseType:"scroll",divId:"hdp-price-history"});

There are several other Ajax calls that pull in other data, tax history, etc that have similar labels.

If it's not obvious already, you'd then need to fetch a secondary URL using http://www.zillow.com/AjaxRender?..." as above, and you'll get the HTML source for the desired data, which can then be scraped directly.

**Comment by hahnicity** [comment]
This sounds like something interesting to do. I don't have much experience working with ajax but from wha

**Comment by glenviewjeff** [comment]
Making progress, extracted the AJAX URLs, now just need to fetch and parse the table.

**Comment by glenviewjeff** [comment]
OK, it really was a pain in the ass.  I committed the price history on my fork.  I'm not a github user, so you'll have to help me out if there's more I need to do to submit a pull request to migrate the changes to your main project.  

It should be pretty trivial to add the tax history, which should also be useful.  I'll do that soon.  

This should be tested on other URLs to make sure it doesn't break.  I only tested it on one house.
